lib/seahorn/BvOpSem2TrackingRawMemMgr.hh

```diff
@@ -12,25 +12,48 @@
 #include <array>
 #include <cstdint>
 
+#include <boost/hana/ext/std/integral_constant.hpp>
+
+#define MAIN_MEM_MGR hana::at_key(m_submgrs, BOOST_HANA_STRING("m_main"))
+
 namespace seahorn {
 namespace details {
 
+/// \brief Container for sub memory managers to be passed to TrackingRawMemMgr
+//
+// Adding a new element (shadow slot) involves adding a new memory manager
+// inside this and in MetadataKind Use of MetadataKind is still needed because
+// setMetadata and getMetadata methods are virtual and thus calls to specific
+// metadata sub managers cannot be resolved at compile-time.
+struct TrackingMemoryTuple {
+  BOOST_HANA_DEFINE_STRUCT(TrackingMemoryTuple, (RawMemManager, m_main),
+                           (RawMemManager, m_r_metadata),
+                           (RawMemManager, m_w_metadata),
+                           (RawMemManager, m_a_metadata));
```

Comment:
to add a new shadow slot, do I just add a new element `(RawMemManager, m_x_metadata)`?